lib/crawler/api/config.rb

```diff
@@ -292,6 +296,30 @@ def validate_domain!(domain)
         raise ArgumentError, "Domain #{domain.inspect} cannot have a path" unless url.path == ''
       end
 
+      def configure_crawl_rules!
+        @crawl_rules = domains.each_with_object({}) do |domain, crawl_rules|
+          url = domain[:url]
+          if domain[:crawl_rules].nil?
+            crawl_rules[url] = {}
+            next
+          end
+
+          raise ArgumentError, "Crawl rules for #{url} is not an array" unless domain[:crawl_rules].is_a?(Array)
+
+          crawl_rules[url] = build_crawl_rules(domain[:crawl_rules], url)
+        end
+      end
+
+      def build_crawl_rules(crawl_rules, url)
+        crawl_rules.map do |crawl_rule|
+          policy = crawl_rule[:policy].to_sym
+          url_pattern = Regexp.new(
+            Crawler::Utils.url_pattern(url, crawl_rule[:type], crawl_rule[:pattern])
+          )
+          Crawler::Data::Rule.new(policy, url_pattern:)
```

Comment:
have I forgotten ruby syntax? The trailing colon looks wrong...

```suggestion
          Crawler::Data::Rule.new(policy, url_pattern: url_pattern)
```

Reply:
In 3.1 syntax you can have has literals with keyword arguments (which also works for assigning key:value pairs to objects). This works as long as the variable's name matches the key. So both my version and your change are the same, but the current linter will prefer my version right now.

Here's some reading on the syntax change if you like: https://rubyreferences.github.io/rubychanges/3.1.html#values-in-hash-literals-and-keyword-arguments-can-be-omitted

Reply:
TIL, thanks! I'd tried to google it, and the SO answer I found must just have been outdated.